tests/test_auth.py

```diff
@@ -1,19 +1,53 @@
+import http.server
+import json
 import logging
 from unittest import mock
 
 import kagglehub
 from kagglehub.config import get_kaggle_credentials
 from tests.fixtures import BaseTestCase
 
+from .utils import create_test_http_server
+
 logger = logging.getLogger(__name__)
 
 
+class KaggleAPIHandler(http.server.BaseHTTPRequestHandler):
+    def do_HEAD(self):  # noqa: N802
+        self.send_response(200)
+
+    def do_GET(self):  # noqa: N802
+        if self.path == "/hello":
+            self.send_response(200)
+            self.send_header("Content-type", "application/json")
+            self.end_headers()
+            self.wfile.write(bytes(json.dumps({"message": "Hello from test server!"}), "utf-8"))
+        else:
+            self.send_response(404)
+            self.send_header("Content-type", "application/text")
+            self.send_header("Content-Length", 0)
+            self.end_headers()
+            self.wfile.write(bytes("", "utf-8"))
+
+
 class TestAuth(BaseTestCase):
     def test_login_updates_global_credentials(self):
```

Comment:
Can you add a test for the 403 (bad credentials) case?

To do this, you can update the do_GET method above.

1. Get the basic auth credentials attached to the request.
1. Compare to the expected good credentials username/key. If they are return 200, otherwise return a 403.

Define the good credentials value in a "constant" at the top of this test file so you can use it in do_GET and in your test case.

Reply:
i added some changes, but not sure how to get the output of kagglehub.login() (the 403 error), with making the login() function raise an exception. let me know what you think

Reply:
I tend to avoid asserting on logs because it makes the tests brittle but you could assert that an ERROR log was emitted (no need to check the content): https://docs.python.org/3/library/unittest.html#unittest.TestCase.assertLogs

WDYT?

Reply:
yeah i think here, this is the only way we can test it currently.